### PR TITLE
DOC: update Contributor Quickstart guide to refer to the Building From Source Guide

### LINKED ISSUE #21421: DOC/DEV: direct to building guide from quickstart guide
gh-21397 reported a missing dependency from [the dev quickstart guide](https://scipy.github.io/devdocs/dev/dev_quickstart.html). Quoting myself from that PR:

> This particular section hasn't been updated since last year, and mostly duplicates the info in [scipy.github.io/devdocs/building/index.html](https://scipy.github.io/devdocs/building/index.html#building-from-source-for-scipy-development), which has been updated more recently. While it is somewhat nice to have a shorter version of the instructions, it does result in more maintenance burden overall, so I think we should just point to the Building From Source guide here.

I would be happy to review a PR which rewords the quickstart guide to direct to https://scipy.github.io/devdocs/building/index.html#building-from-source-for-scipy-development for build instructions, rather than giving them directly.

### PR DESCRIPTION
**Pull request by chitvs**
#### Reference issue
Closes #21421.

#### What does this implement/fix?
Updates the `dev-quickstart` guide to refer directly to the "Building From Source for SciPy Development" guide instead of duplicating the instructions.

#### Additional information
This change helps to reduce maintenance burden and ensures that the information in the `dev-quickstart` guide is consistent with the more up-to-date instructions.

### CODE REVIEW

File: doc/source/dev/dev_quickstart.rst [hunk 1]
```diff
@@ -16,8 +16,9 @@ steps to start contributing:
 2. **Build SciPy**
 
    SciPy uses compiled code for speed, which means you might need extra
-   dependencies to complete this step depending on your system - see
-   :ref:`building-from-source`.
+   dependencies to complete this step depending on your system. For detailed
+   instructions on building SciPy from source, please refer to the 
+   `Building From Source for SciPy Development <https://scipy.github.io/devdocs/building/index.html#building-from-source-for-scipy-development>`_ guide.
```
Comment (tylerjereddy) [inline doc/source/dev/dev_quickstart.rst:21]: Wouldn't this only reduce maintenance burden if we actually delete the older/duplicated instructions about `Conda env` and `Virtual env` below? If I understand Lucas correctly, the point is that we longer want to maintain the extra/older instructions below.

Comment (chitvs) [reply doc/source/dev/dev_quickstart.rst:21]: Alright, got it. So, you want me to remove the duplicated Conda and Virtual environment instructions entirely and just direct people to the Building SciPy From Source guide? Let me know if that's the plan, and I'll make those changes.

Comment (lucascolley) [reply doc/source/dev/dev_quickstart.rst:21]: Yeah, what I had in mind was replacing the section that starts with "execute the
following commands" with a link to the building from source guide. I suppose it is already linked on line 20 though, so maybe there is a better way to restructure the page! The main thing is to remove the conda/venv instructions from the page.

Comment (chitvs) [reply doc/source/dev/dev_quickstart.rst:21]: Ok, there we go.

**Review by lucascolley — APPROVED** [review]
thanks @chitvs, I pushed some minor changes and I'm happy with this now!

@tylerjereddy please squash merge if this reads okay and the doc build looks good!

**Comment by chitvs** [comment]
Hi @tylerjereddy and @lucascolley, is there anything else needed to finalize this PR? Thanks!

Best,
Alessandro (chitvs)

**Comment by lucascolley** [comment]
thanks for your first contribution to SciPy @chitvs !

**Pull request merged**